Add note about not passing 0 vector for cosine sim in k-NN (#6698)
opensearch-project/documentation-website: _search-plugins/knn/painless-functions.md

   ```

   Because scores can only be positive, this script ranks documents with vector fields higher than those without.

With cosine similarity, it is not valid to pass a zero vector (`[0, 0, ...`]) as input. This is because the magnitude of

> such a vector is 0, which would lead to the dreaded `divide by 0` when computing the value. Requests

Reviewer: ```suggestion
such a vector is 0, which raises a `divide by 0` exception when computing the value. Requests
```

> containing the 0 vector will be rejected and a corresponding exception will be thrown.

Reviewer: ```suggestion
containing the zero vector will be rejected and a corresponding exception will be thrown.
```
